WalletWasabi/JsonConverters/ByteArrayJsonConverter.cs

```diff
@@ -29,9 +23,16 @@ public override bool CanConvert(Type objectType)
 		}
 
 		/// <inheritdoc />
-		public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+		public override void WriteJson(JsonWriter writer, byte[]? value, JsonSerializer serializer)
```

Comment:
Why was this change nullable?

Reply:
That is how the base class `JsonConverter<T>` is implemented: https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/JsonConverter.cs#L100-L106

Reply:
I would keep the original behavior: throw an exception in case of null, unless there is a good reason to change.

Reply:
Look, I think null value is not allowed here neither 

https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/JsonConverter.cs#L95

If `T `is not nullable and it is not because it is `byte[]` then it will throw.

Reply:
I added that based on @kiminuo's review https://github.com/zkSNACKs/WalletWasabi/pull/6741#discussion_r756112098

> (When you write JSON, I would expect that you serialize C# null as JSON null. That would be common general behavior.)

Reply:
> Look, I think null value is not allowed here neither
> 
> https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/JsonConverter.cs#L95

This abstract method is implemented in this PR: https://github.com/JamesNK/Newtonsoft.Json/blob/52190a3a3de6ef9a556583cbcb2381073e7197bc/Src/Newtonsoft.Json/JsonConverter.cs#L106

> If `T `is not nullable and it is not because it is `byte[]` then it will throw.

`byte[]` is actually nullable (`byte[]?` is valid syntax).

Also, changing:

```diff
-public override void WriteJson(JsonWriter writer, byte[]? value, JsonSerializer serializer)
+public override void WriteJson(JsonWriter writer, byte[] value, JsonSerializer serializer)
```

leads to a warning by my IDE.


If you are unhappy about this line https://github.com/zkSNACKs/WalletWasabi/pull/6806/files#diff-eaa1419201efc86de9680f2889abe4d23512ec7d78da1794d42883e73f1cf98eR30, we can forbid that by throwing an exception. But the original code was imho incorrect as it did not handle this circumstance.

Reply:
I am fine with the change type change, but the original code threw an exception in case of null - so the error clearly appears. The new code hides the problem.

Reply:
Yeah, sure.

Reply:
@molnard Should I do the same for [other PRs](https://github.com/zkSNACKs/WalletWasabi/pulls?q=is%3Apr+author%3Ayahiheb+is%3Aclosed+from+%27JsonConverter%27+to+%27JsonConverter%3CT%3E%27+) where we allowed null?
@lontivero What do you think?

Reply:
> I am fine with the change type change.

The type didn't change because `byte[]` and `byte[]?` are the same type and c# don't have non-nullable types, the `?` is just a way to ask the code analyzer to follow the value and make sure it is not null

> .... but the original code threw an exception in case of null - so the error clearly appears.

Yes, we have to avoid breaking changes. I agree.

> @lontivero What do you think?

I think we shouldn't change the existing behavior if we are not sure that that is correct. I assumed (wrongly?) that it was writing `null` before but if that's not the case and as @molnard says it was throwing an exception before then it has to throw an exception now too (and the same exception, btw)

Reply:
Done. https://github.com/zkSNACKs/WalletWasabi/pull/6806/commits/4294d6b50cf341e33587f71097aaf1630154c67d